File: fun.py
import discord
import aiohttp
import config
import random
import json
import time
import os
import asyncio
import brfilter
import main
from discord.ext import tasks, commands
brunopowroznik = {0:'https://www.youtube.com/watch?v=6LvlG2dTQKg',1:'https://www.youtube.com/watch?v=ILvd5buCEnU',2:'https://www.youtube.com/watch?v=nEDw_WKeQoc',3:'https://www.youtube.com/watch?v=0YrU9ASVw6w',4:'https://www.youtube.com/watch?v=GxMXWqSauZA',5:'https://www.youtube.com/watch?v=9rtD2omE2N0',6:'https://www.youtube.com/watch?v=-Tqn5NqXskM'}
eightballresponses = {0: 'It is certain.', 1: 'Without a doubt.', 2: 'Yes – definitely.', 3: 'You may rely on it', 4: 'As I see it, yes.', 5: 'Most likely.', 6: 'Outlook good.', 7: 'Yes.', 8: 'Signs point to yes.', 9: 'It is decidedly so.', 10: 'Reply hazy, try again.', 11: 'Ask again later.', 12: 'Better not tell you now.', 13: 'Cannot predict now.', 14: 'Concentrate and ask again.', 15: 'Don\'t count on it.', 16: 'My reply is no.', 17: 'My sources say no.', 18: 'Outlook not so good.', 19: 'Very doubtful.'}

class Fun(commands.Cog):
    def __init__(self, bot):
        self.bot = bot
        self._last_member = None

    @commands.command()
    async def cat(self, ctx):
        """Have a cat picture!"""
        async with aiohttp.ClientSession() as session:
            async with session.get('https://nekos.life/api/v2/img/meow') as r:
                if r.status == 200:
                    js = await r.json()
                    embed = await main.buildEmbed('Here\'s your cat picture!', js['url'])
                    await ctx.send(embed = embed)

    @commands.command(aliases=["8ball"])
    async def eightball(self, ctx, question):
        """Ask me a question!"""
        randomnum = random.randint(0,19)
        await ctx.send(ctx.message.author.nick+' asked the magic 8-ball, **'+str(question)+'**. The magic 8-ball says **'+eightballresponses[randomnum]+'**.')

    @commands.command()
    async def dog(self, ctx):
        """Have a dog picture!"""
        async with aiohttp.ClientSession() as session:
            async with session.get('https://nekos.life/api/v2/img/woof') as r:
                if r.status == 200:
                    js = await r.json()
                    embed = await main.buildEmbed('Here\'s your dog picture!', js['url'])
                    await ctx.send(embed = embed)

    @commands.command()
    async def slap(self, ctx, *, member: discord.Member):
        """Slap somebody around the face!"""
        async with aiohttp.ClientSession() as session:
            async with session.get('https://nekos.life/api/v2/img/slap') as r:
                if r.status == 200:
                    js = await r.json()
                    embed = await main.buildEmbed('{0} slapped {1}!'.format(ctx.message.author.nick, member.nick), js['url'])
                    await ctx.send(embed = embed)

    @commands.command()
    async def hug(self, ctx, *, member: discord.Member):
        """Hug your best friend!"""
        async with aiohttp.ClientSession() as session:
            async with session.get('https://nekos.life/api/v2/img/hug') as r:
                if r.status == 200:
                    js = await r.json()
                    embed = await main.buildEmbed('{0} gave a hug to {1}!'.format(ctx.message.author.nick, member.nick), js['url'])
                    await ctx.send(embed = embed)

    @commands.command()
    async def dice(self, ctx):
        """Roll the dice!"""
        dice = random.randint(1, 6)
        await ctx.send('You rolled the dice and got {}!'.format(dice))

    @commands.command()
    async def feed(self, ctx, *, arg):
        """I am hungry! Feed me!"""
        if '🍆' in arg or 'eggplant' in arg:
            await ctx.send('I don\'t want to bite down on it, I\'d rather suck the juice out of it!')
            await ctx.send('Yum, yum eggplant juice!')
        elif '🍕' in arg or 'pizza' in arg:
            await ctx.send('_lipsmack_ Mmm. I love how creamy the sauce is!')
        elif '🇬🇧' in arg or 'uk flag' in arg:
            await ctx.send('I refuse to eat the UK flag! Unepic!')
        elif '🇩🇪' in arg or 'german flag' in arg:
            await ctx.send('Ich werde die deutsche Flagge nicht essen! Unepisch!')
        elif '🇧🇪' in arg or 'belgium flag' in arg:
            await ctx.send('Hmm. Germany on it\'s side? I\'ll eat it. _rip_ _crunch_ Tastes meh.')
        elif '🏳️‍🌈' in arg or 'rainbow flag' in arg:
            await ctx.send('I- I can\'t eat that flag! I would anger my creator!')
        elif '🍋' in arg or 'lemon' in arg:
            await ctx.send('✝️ KEEP THAT SATANIC YELLOW CITRUS RUTACEAE SAPINDALE ORGANISM AWAY FROM ME!!!! ✝️')
        elif '🍌' in arg or 'banana' in arg:
            await ctx.send('Nice and ripe! _munch munch_')
        elif '🥓' in arg or 'bacon' in arg:
            await ctx.send('Bacon, eggs and bangers and mash!. Now that\'s a traditional english breakfast!')
            await ctx.send('Maybe add some yorkshire pudding.')
        elif '🍑' in arg or 'peach' in arg or 'butt' in arg:
            await ctx.send('_lick_ Can I lick it more?')
        elif '🥵' in arg or 'hot face' in arg or 'moan face' in arg:
            await ctx.send('Feed me your eggplant.')
        elif '🦇' in arg or 'bat' in arg:
            await ctx.send('🤢 Corona! Corona!')
        elif '😳' in arg or 'flushed' in arg:
            await ctx.send('Huh?')
        elif '🥩' in arg or 'meat' in arg:
            await ctx.send('_gets meat hammer_ Time to beat that meat.')
        elif '🐈' in arg or 'cat' in arg or '🐱' in arg:
            await ctx.send('No! You take the pussy!')
        elif '🍄' in arg or 'mushroom' in arg:
            await ctx.send('It\'s a-me, a Drug addict!')
        else:
            await ctx.send('_munch_ I like the taste!')

    @commands.command(pass_context=True)
    async def why(self, ctx):
        """Why?"""
        async with aiohttp.ClientSession() as session:
            async with session.get('https://nekos.life/api/v2/why') as r:
                if r.status == 200:
                    js = await r.json()
                    await ctx.send(js['why'])

    @commands.command()
    async def say(self, ctx, *, arg):
        """Make the bot say stuff."""
        settings_superfilterbans = await main.bot_load_sfbans()
        settings_filter = await main.bot_load_filter()
        if any(s in arg.lower() for s in brfilter.badwords):
            if settings_filter.get(str(ctx.message.guild.id)) == None:
                # print(str(ctx.message.author.id) +' Tried to send ' + str(arg) +' to server ID ' + str(ctx.message.guild.id) + ' with filtering on')
                await ctx.send('Your message contains filtered words!')

            elif settings_filter[str(ctx.message.guild.id)] == 0:
                # print(str(ctx.message.author.id) +' Tried to send ' + str(arg) +' to server ID ' + str(ctx.message.guild.id) + ' with filtering on')
                await ctx.send('Your message contains filtered words!')

            else:
                # print(str(ctx.message.author.id) +' Sent ' + str(arg) +' to server ID ' + str(ctx.message.guild.id) + ' with filtering off')
                await ctx.send(arg)
        elif any(s in arg.lower() for s in brfilter.superbadwords):
            if settings_superfilterbans.get(str(ctx.message.author.id)) == None:
                settings_superfilterbans[str(ctx.message.author.id)] = 0
                msg = ctx.message
                await msg.delete()
                await ctx.send('```Superfilter Alert\nYour message contained super filtered words!\nThe next time you use those, I\'ll have to ban you from this command!```')
                await main.bot_save_sfbans(settings_superfilterbans)
            elif settings_superfilterbans[str(ctx.message.author.id)] == 0:
                settings_superfilterbans[str(ctx.message.author.id)] = 1
                msg = ctx.message
                await msg.delete()
                await ctx.send('```Superfilter Alert\nYour message contained super filtered words!\nYou\'ve been banned from the say command.\nJoin our support server to appeal the ban.```')
                await ctx.send('https://discord.gg/g2SWnrg')
                await main.bot_save_sfbans(settings_superfilterbans)
        elif settings_superfilterbans.get(str(ctx.message.author.id)) != None:
            if settings_superfilterbans[str(ctx.message.author.id)] == 1:
                msg = ctx.message
                await msg.delete()
                await ctx.send('```Superfilter Alert\nYou\'ve been banned from the say command.\nJoin our support server to appeal the ban.```')
                await ctx.send('https://discord.gg/g2SWnrg')
            else:
                await ctx.send(arg)
        else:
            await ctx.send(arg)

    @commands.command()
    async def bruno(self, ctx):
        """A Bruno Powroznik special."""
        settings_filter = await main.bot_load_filter()
        if settings_filter.get(str(ctx.message.guild.id)) == None:
            await ctx.send('This server has filtering enabled.')

        elif settings_filter[str(ctx.message.guild.id)] == 0:
            await ctx.send('This server has filtering enabled.')

        else:
            x = random.randint(0,6)
            await ctx.send(brunopowroznik[x])

def setup(bot):
    bot.add_cog(Fun(bot))

File: Bot/Commands/Fun.cs
// Blu-Ray Discord Bot
//
// Copyright(C) 2020 Colean, Apfel
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program. If not, see <http://www.gnu.org/licenses/>.

using Bot.Structures;
using DSharpPlus.CommandsNext;
using DSharpPlus.CommandsNext.Attributes;
using DSharpPlus.Entities;
using System;
using System.Threading.Tasks;

namespace Bot.Commands
{
    public class Fun : BaseCommandModule
    {
        private readonly string[] eightballResponses =
        {
            "It is certain",
            "Without a doubt",
            "Yes – definitely",
            "You may rely on it",
            "As I see it",
            "Most likely",
            "Outlook good",
            "Yes",
            "Signs point to yes",
            "It is decidedly so",
            "Reply hazy",
            "Ask again later",
            "Better not tell you now",
            "Cannot predict now",
            "Concentrate and ask again",
            "Don\"t count on it",
            "My reply is no",
            "My sources say no",
            "Outlook not so good",
            "Very doubtful"
        };

        private readonly string[] brunoVideos =
        {
            "https://www.youtube.com/watch?v=6LvlG2dTQKg",
            "https://www.youtube.com/watch?v=ILvd5buCEnU",
            "https://www.youtube.com/watch?v=nEDw_WKeQoc",
            "https://www.youtube.com/watch?v=0YrU9ASVw6w",
            "https://www.youtube.com/watch?v=GxMXWqSauZA",
            "https://www.youtube.com/watch?v=9rtD2omE2N0",
            "https://www.youtube.com/watch?v=-Tqn5NqXskM"
        };

        [Command("8ball"), Description("Ask the magic 8-Ball a question."), Aliases("eightball", "oracle")]
        public async Task EightballAsync(CommandContext context, [Description("The question to ask the 8-Ball about."), RemainingText] string question) => await context.RespondAsync(embed: new DiscordEmbedBuilder()
            .WithAuthor($"{context.User.Username}", iconUrl: context.User.AvatarUrl)
            .WithDescription($"You've asked the magic 8-Ball the following question:\n```\n{question}\n```\nMy Answer is: **{eightballResponses[new Random().Next(0, eightballResponses.Length - 1)]}**."));

        [Command("bruno"), Description("Shows you a random Bruno Powroznik video."), Aliases("powroznik")]
        public async Task BrunoAsync(CommandContext context) => await context.RespondAsync($"Here's a Bruno Powroznik video for you:\n\n{brunoVideos[new Random().Next(0, brunoVideos.Length - 1)]}");

        [Command("cat"), Description("Shows a cute picture of a cat."), Aliases("meow")]
        public async Task CatAsync(CommandContext context) => await context.RespondAsync(embed: new DiscordEmbedBuilder().WithTitle("Here's your cat picture!").WithImageUrl(await NekosLifeImage.GetAsync(NekosLifeImage.Endpoint.Meow)));

        [Command("say"), Description("Make the bot say something funny!")]
        public async Task SayAsync(CommandContext context, [Description("Make the bot say something funny!"), RemainingText] string text) => await context.RespondAsync($"{text}");

        [Command("dice"), Description("Rolls a dice of a range from 1 to 6."), Aliases("roll")]
        public async Task DiceAsync(CommandContext context) => await context.RespondAsync($"You rolled the dice. You got a **{new Random().Next(1, 6)}**.");

        [Command("dog"), Description("Shows a cute picture of a dog."), Aliases("woof")]
        public async Task DogAsync(CommandContext context) => await context.RespondAsync(embed: new DiscordEmbedBuilder().WithTitle("Here's your dog picture!").WithImageUrl(await NekosLifeImage.GetAsync(NekosLifeImage.Endpoint.Woof)));

        [Command("hug"), Description("Hugs another member."), Aliases("squeeze"), RequireGuild]
        public async Task HugAsync(CommandContext context, [RemainingText, Description("Member to hug.")] DiscordMember member = null)
        {
            if (member == null)
            {
                await context.RespondAsync("You need to name someone to hug.");
                return;
            }

            await context.RespondAsync(embed: new DiscordEmbedBuilder().WithTitle($"{context.User.Username} hugged {member.Username}.").WithImageUrl(await NekosLifeImage.GetAsync(NekosLifeImage.Endpoint.Hug)));
        }

        [Command("kiss"), Description("Kisses another member."), Aliases("smooch"), RequireGuild]
        public async Task KissAsync(CommandContext context, [RemainingText, Description("Member to kiss.")] DiscordMember member = null)
        {
            if (member == null)
            {
                await context.RespondAsync("You need to name someone to kiss.");
                return;
            }

            await context.RespondAsync(embed: new DiscordEmbedBuilder().WithTitle($"{context.User.Username} kissed {member.Username}.").WithImageUrl(await NekosLifeImage.GetAsync(NekosLifeImage.Endpoint.Kiss)));
        }

        [Command("pat"), Description("Hugs another member."), Aliases("pet", "headpat"), RequireGuild]
        public async Task PatAsync(CommandContext context, [RemainingText, Description("Member to pat on the head.")] DiscordMember member = null)
        {
            if (member == null)
            {
                await context.RespondAsync("You need to name someone to pat on their head.");
                return;
            }

            await context.RespondAsync(embed: new DiscordEmbedBuilder().WithTitle($"{context.User.Username} pat {member.Username} on their head.").WithImageUrl(await NekosLifeImage.GetAsync(NekosLifeImage.Endpoint.Pat)));
        }

        [Command("slap"), Description("Slaps another member."), Aliases("hit"), RequireGuild]
        public async Task SlapAsync(CommandContext context, [RemainingText, Description("Member to slap.")] DiscordMember member = null)
        {
            if (member == null)
            {
                await context.RespondAsync("You need to name someone to slap.");
                return;
            }

            await context.RespondAsync(embed: new DiscordEmbedBuilder().WithTitle($"{context.User.Username} slapped {member.Username}.").WithImageUrl(await NekosLifeImage.GetAsync(NekosLifeImage.Endpoint.Slap)));
        }

        [Command("why"), Description("Gives you a random question."), Aliases("what")]
        public async Task WhyAsync(CommandContext context)
        {
            string question = await NekosLifeWhyQuestion.GetAsync();
            await context.RespondAsync($"Here's a question for you:\n\n{char.ToUpper(question[0]) + question.Substring(1)}");
        }
    }
}
